mcs/class/corlib/System/ArgIterator.cs

```diff
@@ -86,22 +86,30 @@ public TypedReference GetNextArg ()
 		{
 			if (num_args == next_arg)
 				throw new InvalidOperationException (Locale.GetText ("Invalid iterator position."));
-			return IntGetNextArg ();
+			TypedReference result = new TypedReference ();
+			unsafe {
+				IntGetNextArg (&result);
+			}
+			return result;
 		}
 
 		[MethodImpl (MethodImplOptions.InternalCall)]
-		extern TypedReference IntGetNextArg ();
+		extern unsafe void IntGetNextArg (void *res);
```

Comment:
Shouldn't this have used `ref TypedReference res` to make it "safe" code.

Reply:
The corefx/coreclr code uses this convention for some reason.